### ISSUE TITLE
TypeError: no implicit conversion of nil into String

### ISSUE THREAD

**Issue by traylenator** [rodjek/puppet-lint#830]
---
puppet-lint version: 2.3.6
ruby version: 2.2.9-p480
platform: x86_64-linux
file path: manifests/private/test_case.pp
file contents:
```
define hg_punch::private::test_case {

  $exclusive_envs = ['foo','bar']

  if $exclusive_envs {
    $_deny_env_params = {
      'http-request' => "deny unless { ${join($exclusive_envs.map |$_env| {"urlp(environment) ${_env}"}, ' or ')} }" ,
    }
  } else {
    $_deny_env_params = {}
  }

  
}

```
error:
```
TypeError: no implicit conversion of nil into String
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer.rb:321:in `+'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer.rb:321:in `slurp_string'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer.rb:226:in `tokenise'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer.rb:473:in `interpolate_string'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer.rb:228:in `tokenise'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:25:in `load_data'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:55:in `run'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint.rb:205:in `run'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/tasks/puppet-lint.rb:83:in `block (3 levels) in define'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/tasks/puppet-lint.rb:81:in `each'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/tasks/puppet-lint.rb:81:in `block (2 levels) in define'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/file_utils_ext.rb:59:in `verbose'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/puppet-lint-2.3.6/lib/puppet-lint/tasks/puppet-lint.rb:75:in `block in define'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:271:in `call'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:271:in `block in execute'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:271:in `each'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:271:in `execute'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:213:in `block in invoke_with_call_chain'
/opt/rh/rh-ruby22/root/usr/share/ruby/monitor.rb:211:in `mon_synchronize'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:193:in `invoke_with_call_chain'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:237:in `block in invoke_prerequisites'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:235:in `each'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:235:in `invoke_prerequisites'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:212:in `block in invoke_with_call_chain'
/opt/rh/rh-ruby22/root/usr/share/ruby/monitor.rb:211:in `mon_synchronize'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:193:in `invoke_with_call_chain'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/task.rb:182:in `invoke'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:160:in `invoke_task'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:116:in `each'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:116:in `block in top_level'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:125:in `run_with_threads'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:110:in `top_level'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:83:in `block in run'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:186:in `standard_exception_handling'
/afs/cern.ch/user/s/straylen/.gem/ruby/gems/rake-12.3.1/lib/rake/application.rb:80:in `run'
/opt/rh/rh-ruby22/root/usr/bin/rake:33:in `<main>'
```

**Comment by camilo-schoeningh-sociomantic** [comment]
We have the same issue. Does the linter have a option to disable all checks in a block ? I could not find it.

**Comment by rodjek** [comment]
Fixed by #846
